lib/auth/auth.go

```diff
@@ -1889,21 +1846,27 @@ func (a *Server) RegisterUsingToken(req RegisterUsingTokenRequest) (*PackedKeys,
 	}
 
 	// generate and return host certificate and keys
-	keys, err := a.GenerateServerKeys(GenerateServerKeysRequest{
-		HostID:               req.HostID,
-		NodeName:             req.NodeName,
-		Roles:                types.SystemRoles{req.Role},
-		AdditionalPrincipals: req.AdditionalPrincipals,
-		PublicTLSKey:         req.PublicTLSKey,
-		PublicSSHKey:         req.PublicSSHKey,
-		RemoteAddr:           req.RemoteAddr,
-		DNSNames:             req.DNSNames,
-	})
+	certs, err := a.GenerateServerKeys(context.Background(),
+		&proto.GenerateServerKeysRequest{
+			HostID:               req.HostID,
+			NodeName:             req.NodeName,
+			Roles:                []string{string(req.Role)},
+			AdditionalPrincipals: req.AdditionalPrincipals,
+			PublicTLSKey:         req.PublicTLSKey,
+			PublicSSHKey:         req.PublicSSHKey,
+			RemoteAddr:           req.RemoteAddr,
+			DNSNames:             req.DNSNames,
+		})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	log.Infof("Node %q [%v] has joined the cluster.", req.NodeName, req.HostID)
-	return keys, nil
+	return &PackedKeys{
```

Comment:
The only difference at this point between `auth.PackedKeys` and `proto.Certs` is that `PackedKeys` has a field for the private key. We don't populate that field here since the auth server doesn't have the private key, so I'm looking for thoughts on whether we should just return `proto.Certs` here as well.

Reply:
If we never return the private keys, then returning proto.Certs seems to be more appropriate.